tests/src/kakarot/accounts/test_contract_account.py

```diff
@@ -123,3 +123,40 @@ def test_should_read_bytecode(self, cairo_run, bytecode, storage):
             calls = [call(address=address) for address in addresses]
             mock_storage.assert_has_calls(calls)
             assert output[:output_len] == list(bytecode)
+
+    class TestUpgrade:
+        @SyscallHandler.patch(
+            "IAccount.library_call_version", lambda class_hash, data: [2000]
+        )
+        def test_should_upgrade_account(self, cairo_run):
+            cairo_run("test__upgrade", new_class=0xBEEF)
+
+            SyscallHandler.mock_library_call.assert_any_call(
+                class_hash=0xBEEF,
+                function_selector=get_selector_from_name("version"),
+                calldata=[],
+            )
+            SyscallHandler.mock_replace_class.assert_called_once_with(
+                address=0xABDE1, class_hash=0xBEEF
+            )
+            SyscallHandler.mock_storage.assert_called_once_with(
+                address=get_storage_var_address("Account_implementation"), value=0xBEEF
+            )
+
+        # TODO: Does not work
+        # assert SyscallHandler.class_hash == 0xbeef
```

Comment:
doesn't pass because you are update a particular instance attribute, and not a class attribute. to me, the `assert_called_once_with` is enough, it's SNOS responsibility to actually replace the class hash

Reply:
makes sense, yes